Replace errors package with github.com/pkg/errors

ansiwen opened the pull request:
### What does this PR achieve? Why do we need it?

The errors we return and handle are forwarded through many levels of the function call stack, so often -- especially while debugging -- it can be very hard to see, where an error really is originating from. This is especially true if errors are forwarded through channels to different goroutines or asynchronously from 3rd party libraries.

In order to improve this situation the advice of [this](https://dave.cheney.net/2016/06/12/stack-traces-and-the-errors-package) blog article has been followed. That is the [github.com/pkg/errors](https://godoc.org/github.com/pkg/errors) package is used instead of the standard errors package throughout the code base, which annotates all errors with a stack trace, and also allows to wrap errors with new stack traces, in case of "stack-switches", like when they are transferred asynchronously over channels or come in from 3rd party libraries. The only "old" errors that are kept are the sentinel errors defined in apps/glusterfs/errors.go.

phlogistonjohn commented on apps/glusterfs/errors.go:
```diff
@@ -10,26 +10,26 @@
 
 import (
-	"errors"
+	stdErrors "errors"
```
I don't understand this change. There's no import of the enhanced errors package here.

ansiwen replied:
I like to make it explicit here, that this is the only place where still the "old" errors package is used. Also it avoids, that when code is added to the file (like I did in follow up PRs) doesn't accidentally use the standard errors package.

phlogistonjohn commented:
Looks pretty good so far. A couple of points:

First, this is somewhat of a nit and I don't think I'd block the merge over this but I like a style of imports like:
```golang
import (
// std library imports go first
   "fmt"
   "io"
   "strings"

// non-local library imports go next
   "github.com/something/amazing"
   "github.com/cool/stuff/pkg/foobar"

// local (within the current repo last)
    "github.com/heketi/heketi/pkg/womble"
)
```

Some code in heketi follows this while some code doesn't. 

Second, like any change that touches glide I think we need to argue about the delta in glide lock. Can I assume that most of these changes to all the other imports were done automatically by glide?

ansiwen commented:
@phlogistonjohn: Isn't the import style enforced by the format tool? I didn't do anything on purpose here and just let the tool do it's work.

phlogistonjohn commented:
> Isn't the import style enforced by the format tool? I didn't do anything on purpose here and just let the tool do its work.

The tool only enforces that each group of import lines is in sorted order. Logically grouping them by separating them by empty lines is a convention I have seen elsewhere and liked.

phlogistonjohn commented:
@ansiwen needs rebase & conflict resolution

ansiwen commented:
@phlogistonjohn I rebased, removed the stdErrors stuff and reorganized the import lines. And yes, the glide.lock changes were done automatically by `glide update -v`

phlogistonjohn commented:
Thank you for your contribution however I think we'll decline taking this approach. I learned a lot from this PR but I think it's a bit too much change for such a fundamental part of heketi. If we were to fundamentally change our error handling I'd prefer we look into the new native approach in Go 1.13.
